src/main/kotlin/nl/nuts/discovery/store/entity/Certificate.kt

```diff
@@ -49,28 +44,26 @@ class Certificate {
             return Certificate().apply {
                 name = certificate.subjectDN.name
                 x509 = certificate.encoded
-                oid = extractOID(certificate)
+                oid = getSinglePartyID(certificate)
                 ca = caSubject
                 this.chain = chain
             }
         }
 
-        /**
-         * Find Nuts identifier in extensions
-         */
-        fun extractOID(certificate: X509Certificate): String? {
-            val sans = JcaX509ExtensionUtils.getSubjectAlternativeNames(certificate)
-            sans.forEach {// GeneralNames
-                val generalName = it as List<*> // GeneralName
-                if (generalName[0] == GeneralName.otherName) {
-                    // then this should be a DLSequence
-                    val seq = generalName[1] as DLSequence
-                    if (seq.getObjectAt(0) == NutsCertificateRequest.NUTS_VENDOR_EXTENSION) {
-                        return (seq.getObjectAt(1) as DERUTF8String).toString()
+        private fun getSinglePartyID(certificate: X509Certificate): PartyId? {
+            return certificate.subjectAlternativeNames
+                    // Take SANs of type OtherName
+                    .filter { it[0] == 0 }
```

Comment:
magic number

Reply:
Found a sweet constant for it (`GeneralName.otherName`)